app/uk/gov/hmrc/selfservicetimetopay/controllers/CalculatorController.scala

```diff
@@ -34,9 +34,57 @@ class CalculatorController(calculatorConnector: CalculatorConnector) extends Tim
     Redirect(routes.CalculatorController.getAmountsDue())
   }
 
+  def getDebitDate: Action[AnyContent] = Action.async { implicit request =>
+    sessionCache.get.map[Result] {
+      case Some(TTPSubmission(_, _, _, _, Some(EligibilityTypeOfTax(true, false)), Some(EligibilityExistingTTP(Some(false))), _, _, _, _)) =>
```

Comment:
We need to look at possibly setting the date to None on this page load due to navigational issues